parlai/core/logs.py

```diff
@@ -234,6 +246,17 @@ def log_final(self, setting, report):
         }
         for key, value in report.items():
             self.run.summary[key] = value
+        if self.model_file is not None and self.log_model:
```

Comment:
Not sure if `log_final` is called at the end of each epoch or only after training. But if it's called on epoch end, we should ideally have an argument to control the frequency of logging the model artifact. Something like checkpointing at intervals or checkpointing when a criterion is met; can also go with logging the best checkpoint at the end with an alias best.

Reply:
Changed this to log artifact on every successful save